lib/Tcl/Tk.pm

```diff
@@ -1,6 +1,7 @@
 package Tcl::Tk;
 
 use strict;
+use 5.006;
```

Comment:
the perl version already checked during Makefile.PL stage, I think adding this extra check - to avoid haskers who were brave enough to bypass Makefile.PL check??

Reply:
As far as I understand it, the Makefile.PL just sets metadata and doesn't actually enforce anything (other than allowing things like `cpanm` and CPANTS to work automatically).  I took my inspiration for the patch from http://blogs.perl.org/users/neilb/2014/08/specify-the-min-perl-version-for-your-distribution.html where Neil Bowers mentions that to set the minimum Perl version, one sets it at the top of the module.  If you don't want the change, I'm OK with it; I just wanted to make sure my patch contained all the elements that seem to be deemed necessary by current best practices.

Reply:
point is that module will even probably work at 5.002, I hope :)

Reply:
also, your link namely mentions "makefile.pl" specifying perl version. 
It is present in Tcl::Tk, isn't it?

Reply:
what exact place in Tk.pm "too modern" to request namely 5.006, and not version below?

Reply:
IOW. the statement 'require 5.006' is not precise enough, at actually requires perl version even before that

Reply:
Sorry, I didn't want to be unclear or to push you to do something you don't want to.  In Neil's blog post he specifically says to put the Perl version in the code (in the "Specifying the min perl version in your code") section.  He also mentions that this information should be put in the metadata, i.e. in the Makefile.PL.  Hence (as far as I understand it), one needs to mention the version in both places.  At the very least, the version in the code is what is required so that `perlver` finds an explicitly-set version number.

Upon running `perlver` (from [Perl::MinimumVersion](https://metacpan.org/pod/Perl::MinimumVersion) I noticed that the syntax of `lib/Tcl/Tk.pm` (e.g. the substitution on line 865) and `tk-demos/widget_lib/mkTxtSearch.pl` require Perl 5.5.  Upon looking in the `Makefile.PL` I noticed that the minimum version had been set to 5.6 so guessed that this was the version that you really wanted to have set for the dist.  If you want, maybe I can try to test the module on Perl 5.2 and see if it works and maybe the minimum Perl version restriction can be reduced.

Reply:
BTW: thank you for checking my patches so thoroughly; this definitely helps me be more careful when creating and submitting patches in the future :-)

Reply:
actually "tcl.pm" "requires" 5.6 probably to support unicode, and also probably have something in XS which was introduced in 5.6 (IDK actually).

Tcl::Tk however is fully version agnostic, so I would prefer to remove version requirement at all.

given that you've discovered feature that requires 5.5, I would prefer to even lower version requirement in Makefile.PL.

despite of fact that dependent module requires 5.6, so it will use 5.6 anyway.

btw it isn't even listed in
https://cpants.cpanauthors.org/release/VKON/Tcl-Tk-1.29

as you may have noticed, I am not a great fan of placing version of it, but - to honor yuor mentioned link - I could agree on placing 

"use 5;"
just to underline that I am not stealing any freedom from user and to make some amount of fun out of it :):):):)

Reply:
... I just checked that "use 5; " works fine on 5.004_02

Reply:
Ok, I'll give that a go and push a fix soon.

Reply:
Actually, since the minimum version required by the syntax in `lib/Tcl/Tk.pm` is version 5.5, shouldn't the explicit minimum version also be set to 5.5?  Sorry for being a pain in the backside about this; just want to get the patch right in the end.

Reply:
OK, I'll set it to `use 5;` and push in a minute.

Reply:
I think there was a misunderstanding of what the recommendation was.

The `use 5.x;` statement should only specify a version that is equal or higher to the version required, and **not** a version lower than required.

Because `use 5;` (i.e. Perl 5.0) is now specified but Perl 5.5 syntax is present, `perlver lib/Tcl/Tk.pm` now outputs `ERROR   : ACTUAL DEPENDENCY HIGHER THAN SPECIFIED` (use `perlver --blame` for finding syntax responsible).

I think a decent attempt to address this was already made, so I will forgo proposing to change this.

Reply:
ok, show me what is of '5.5' syntax

...and I am more about to delete this `use 5` at all, rather than figuring out exact perl version that is required. probably 5.6. but - why care? To please buggy linter?

Reply:
The 5.5 syntax in lib/Tcl/Tk.pm is the use of negative lookbehind `(?<!_)` (see https://github.com/Perl/perl5/blob/perl-5.005/pod/perldelta.pod#regular-expressions)

Running `perlver --blame lib/Tcl/Tk.pm` repeatedly:

```
------------------------------------------------------------
 File    : lib/Tcl/Tk.pm
 Line    : 866
 Char    : 2
 Rule    : _regex
 Version : 5.005
 s/(?<!_)__(?!_)/::/g
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : lib/Tcl/Tk.pm
 Line    : 867
 Char    : 2
 Rule    : _regex
 Version : 5.005
 s/(?<!_)___(?!_)/_/g
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : .\lib\Tcl\Tk.pm
 Line    : 2360
 Char    : 2
 Rule    : _regex
 Version : 5.005
 ------------------------------------------------------------
 s/(?<!_)__(?!_)/::/g
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : .\lib\Tcl\Tk.pm
 Line    : 2361
 Char    : 2
 Rule    : _regex
 Version : 5.005
 ------------------------------------------------------------
 s/(?<!_)___(?!_)/_/g
 ------------------------------------------------------------
```

`perlver` doesn't see anything else it knows a minimum version for in lib/Tcl/Tk.pm.

Reply:
hm... ok, probably I will remove this negative lookbehind.
thank you